pkgs/by-name/ob/objdiff/package.nix

```
@@ -0,0 +1,89 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+  stdenv,
+  pkg-config,
+  autoPatchelfHook,
+  darwin,
+  fontconfig,
+  wayland,
+  libxkbcommon,
+  libGL,
+  libX11,
+  libXcursor,
+  libXi,
+  zenity,
+}:
+let
+  version = "2.0.0";
+  commit = "3846a7d315a9d1e3ffaf3f917345ffb225b1b400";
+in
+rustPlatform.buildRustPackage {
+  pname = "objdiff";
+  inherit version;
+
+  src = fetchFromGitHub {
+    owner = "encounter";
+    repo = "objdiff";
+    rev = "v${version}";
+    hash = "sha256-iheRHnmoX0lXAWL3t7HKN/BbcPyup0euom3Jw3kQlVI=";
+  };
+
+  # objdiff-cli doesn't use vergen, but its own bespoke invocation of the git
+  # CLI. Remove the entire build script, and provide the data ourselves.
+  postPatch = ''
+    rm objdiff-cli/build.rs
+    substituteInPlace objdiff-cli/Cargo.toml \
+      --replace-fail 'build = "build.rs"' ""
+  '';
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+      "notify-6.1.1" = "sha256-g1IIpU9Htmv6Gzb70MuSGtji8pGZG6WomyiZfOHSpjA=";
+    };
+  };
+
+  nativeBuildInputs = lib.optionals (!stdenv.isDarwin) [
+    pkg-config
+    autoPatchelfHook
+  ];
+
+  buildInputs = if stdenv.isDarwin then [ darwin.apple_sdk.frameworks.AppKit ] else [ fontconfig ];
```

```suggestion
  buildInputs = if stdenv.hostPlatform.isDarwin then [ darwin.apple_sdk.frameworks.AppKit ] else [ fontconfig ];
```
Same for other checks in the file.